### PR TITLE
Ben upload and ingest for Github links

### LINKED ISSUE #19: Support Uploading a SINGLE Zipped File Package [BE]
## Task Information
Allow anyone to upload one single zipped file package at a time. 

**Notes**
- Keep in mind, there will eventually be an update to this feature, limiting access based on USD permissions. 
- However, that is not your responsibility to write. That is probably assigned to someone else.
- Please instead just focus on the basic case of allowing any users to access this functionality. 

## Subtasks
[ ] Create any necessary classes (e.g. a class for zipped file packages if it doesn't already exist, etc.)
[ ] Store the package info within the package database via query function call
[ ] Communicate response to the FE

### LINKED ISSUE #29: Add Debloat Option to UPLOAD [BE]
## Task Information
Add a debloat option to upload to remove unnecessary bloat from the package
- Cf. tree shaking, minification, etc.

**Notes**
- Keep in mind, there will eventually be an update to this feature, limiting access based on USD permissions. 
- However, that is not your responsibility to write. That is probably assigned to someone else.
- Please instead just focus on the basic case of allowing any users to access this functionality. 
- This feature should not damage the original record of the requested package within our database

### LINKED ISSUE #41: Support Ingestion of Public npm Packages [BE]
## Task Information
Enable requests for the ingestion of a public npm package
- to be ingestible, must be >= 0.5 on each of the non-latency behavior scores
- obviously this implies that a compatible license is required

The PRMergeControl score comes from the following: 
The fraction of project code that was introduced through pull requests with a code review

  
**Notes**
-

### PR DESCRIPTION
**Pull request by bkanter-purdue**
addresses #41 
Closes #29 
Closes #19

### CODE REVIEW

**Review by BellJrDev — CHANGES_REQUESTED** [review]
Before code review will take place, some changes need to be made to the PR structurally

1. Separate upload and ingest into 2 different PRs. PRs should not include multiple full features that have dependency on each other (i.e. ingest fully depends on upload, so upload should be PR'd separately. Upon approval, ingest can be reviewed.)

2. Fix the linking on the PR. Debloat has already been marked completed and closed, so there should not be another PR that claims to close it. If you are not done with the ingest feature, it should not have a PR yet. That being said: there should not be anything more to add to the functionality if it supports ingestion via GitHub link. For all of our purposes, all package ratings will be based solely on the associated GitHub link. Thus, ingestion should just always use the GitHub repo matching npm packages to upload to our registry. With this information, (A) the new Ingest PR should **close** that feature. or (B) if that functionality is not already there, no PR should be opened for ingest yet.

**Comment by BellJrDev** [comment]
Also, GitHub has indicated conflicts that must be solved in this branch's code. 
image

All of these must be resolved before any approval can be granted. Probably start by pulling dev and merging branch(es) with it

**Comment by bkanter-purdue** [comment]
Forgot to merge dev in this morning. Merged it yesterday while working on it. Apologies for that. 
Updated the import paths

As for Debloat, according to my task list, it is not complete since it depended on Upload, so I left it open. Debloat was separated into two tasks, but it was only one piece of core logic that I worked with Tim that needed to be used in two different places. When I worked with Tim, I created the Debloat function in a way where it can be plugged in anywhere.  Debloat on update is complete. This closes debloat on upload. 

As we discussed in the Discord, NPM links do not need to be supported by Upload, so these two commits will bring ingest to a complete status. Built the structure of the code to support both just in case. Simplifies testing process being able to remove it. 

Working on resolving tests. 
Closes #41

File: BE/src/Types/ResponseTypes/index.d.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ declare module "ResponseTypes" {
     export interface DeletePackageViaIDResponse extends Response {}
 
     export type UploadInjestResponseMessages =
-        | "There is missing field(s) in the PackageData or it is formed improperly (e.g. Content and URL ar both set)"
+        | "There is missing field(s) in the PackageData or it is formed improperly (e.g. Content and URL are both set)"
```
Comment (j-leidy) [inline BE/src/Types/ResponseTypes/index.d.ts:64]: thank u

**Comment by j-leidy** [comment]
Only one question. From what I see here: 
image
I am unsure if we need to support the npm links. It looks like maybe you and Dorian had a discussion on this.
If we are certain we don't need to support them this PR is ready to merge.

**Comment by j-leidy** [comment]
Everything works as it should. Excellent work!

**Review by j-leidy — APPROVED** [review]
Excellent work!

**Review by BellJrDev — APPROVED** [review]
John Leidy said to approve it

**Pull request merged**